### ISSUE TITLE
export cleanup should regenerate index.txt

### ISSUE THREAD

**Issue by mikehelmick** [google/exposure-notifications-server#1148]
### TL;DR

When the cleanup job runs, it deletes from the database, and then removes from storage old files.

Those old files continue to appear in the index.txt until the next export runs.

**Expected behavior**

All files that appear in index.txt are not deleted by our application.

**Suggested fixes**

2 possible fixes.

1) determine which files are to be deleted. regnerate the index.txt file without those files. then delete the files.

2) move cleanup to more of a mark and sweep. after X days, mark the file as stale in the database. modify export index generation to skip stale files. after X+Y days do the normal cleanup.

The first path is more correct, the second path is possibly easier to implement.

/priority important-soon

**Comment by jeremyfaller** [comment]
As I think about this, method (2) seems more correct than method (1), which seems to me to have a race condition. That race condition is:

- Client downloads index file.
- Client begins downloading files.
- Cleanup job is triggered, causing the some of the files to be deleted.
- Client tries to download a file that's been deleted, triggering an error.

Strategy (2) doesn't completely eliminate this race condition (if a client takes a really long time to download files, or uses a stale index), but it at least mitigates it.

Without other input, I'm going to implement strategy (2).

**Comment by mikehelmick** [comment]
agreed - the mark and seep is a better approach
latency between the two doesn't have to be long - O(hours)
